daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/generators/CodeGeneratorState.scala

```diff
@@ -262,6 +288,18 @@ class CodeGeneratorState {
       structs.top.parserStatements += parseStatement
       structs.top.unparserStatements += unparseStatement
     }
+
+    // Implement padding if complex type has an explicit length
+    if (context.maybeFixedLengthInBits.isDefined && context.maybeFixedLengthInBits.get > 0) {
+      val octalFillByte = context.fillByteEv.constValue.toByte.toOctalString
+      val parseStatement = s"    parse_fill_bytes(end_position, pstate);"
+      val unparseStatement =
+        s"""    ustate->fill_byte = '\\$octalFillByte';
+           |    unparse_fill_bytes(end_position, ustate);""".stripMargin
+
```

Comment:
We should try to learn from the Scala back-end experience here.

We tried an approach in the Scala back-end based on 
(a) things like fill byte change infrequently
(b) so we set them into the state when they change
(c) processors use them from the state
This can be made to work, but for the scala back-end out-of-order streaming stuff, requires copying this piece of state when suspended. Had this information been in the ERD/RD structure, no copying would be needed. 

Honestly I think anything that can be done via storing a value in the ERD instead is better done that way. It's simpler, not slower - you are still one indirection away from the required value. Basically, processors should be parameterized by an ERD (or other RunitmeData object, e.g. ChoiceRuntimeData), and a State. Anything that can be in the ERD should be. Only the other things should be in the state.